src/core/Bootstrapper.js

```diff
@@ -0,0 +1,37 @@
+/* eslint-disable no-plusplus */
+/* eslint no-await-in-loop: 0 */
+
+export default class Bootstrapper {
+
+  /**
+     *
+     * @param {RCISDKConfig} config
+     */
+  constructor (config, transport, producer) {
+    this.config = config;
+    this.transport = this.provisionTransport(transport);
+    this.producer = this.provisionProducer(producer);
+  }
+
+  provisionTransport (Transport) {
+    let input;
+    if ((Array.isArray(this.config.tenancies) && this.config.tenancies.length > 0) && ((typeof this.config.targetUrl === 'string' || this.config.targetUrl instanceof String) && this.config.targetUrl.length > 0)) {
```

Comment:
@robertdumitrescu MAINTAINABILITY - This is a subtly shifting contract.  Please be sure to make sure it is well documented!

Reply:
OK, makes sense, where shall it be documented? I already added a jsdoc spec for it.

Reply:
Looks good